### PR TITLE
[TypeScript docs] document WorkflowService advanced usage

### PR DESCRIPTION
**Pull request by sw-yx**
## What does this PR do?

document WorkflowService advanced usage

### CODE REVIEW

File: docs/typescript/clients.md [hunk 1]
```diff
@@ -154,6 +155,7 @@ const result = await handle.result(); // block until the workflow completes, if
 Using a Workflow Handle isn't necessary with `client.execute` by definition.
 
 - **Don't forget to handle errors here** - if you call `result()` on a Workflow that prematurely ended for some reason, it will [throw an Error](https://typescript.temporal.io/api/classes/client.WorkflowFailedError) reflecting that reason.
+  `ts const handle = client.getHandle(workflowId); try { const result = await handle.result(); // block until the workflow completes, if you wish } catch (err) { if (err instanceOf WorkflowFailedError) { throw new Error('Temporal workflow failed: ' + workflowId), err) } else { throw new Error('error from Temporal workflow ' + workflowId), err) } } `
```
Comment (sw-yx) [inline docs/typescript/clients.md:158]:
```suggestion
  ```ts
   const handle = client.getHandle(workflowId);
   try {
    	const result = await handle.result(); // block until the workflow completes, if you wish 
  	} catch (err) {
  	  if (err instanceOf WorkflowFailedError) {
  	     throw new Error('Temporal workflow failed: ' + workflowId), err)
      } else {
         throw new Error('error from Temporal workflow ' + workflowId), err)
    }
   } 
   ```
```

File: docs/typescript/clients.md [hunk 2]
```diff
@@ -246,3 +259,31 @@ export async function CancelExternalWorkflow(wfId: string): void {
...
+// alternate Node callback-style API
+connection.service.startWorkflowExecution(/* options */, callbackFn);
+```
+
+Using gRPC calls is often the only way to access some of the more advanced queries you can make from Temporal Server.
+We highlight some queries of interest here:
+
+- [describeTaskQueue](https://typescript.temporal.io/api/classes/proto.temporal.api.workflowservice.v1.WorkflowService-1#describetaskqueue)
+- [getWorkflowExecutionHistory](https://typescript.temporal.io/api/classes/proto.temporal.api.workflowservice.v1.WorkflowService-1#getworkflowexecutionhistory)
+- [listWorkflowExecutions](https://typescript.temporal.io/api/classes/proto.temporal.api.workflowservice.v1.WorkflowService-1#listworkflowexecutions)
```
Comment (sw-yx) [inline docs/typescript/clients.md:289]:
```suggestion
- [listWorkflowExecutions](https://typescript.temporal.io/api/classes/proto.temporal.api.workflowservice.v1.WorkflowService-1#listworkflowexecutions)

```ts
  const res = await connection.service.listWorkflowExecutions({ namespace: 'default'})
  console.table(res.executions)
```

Outputs:

```

┌─────────┬───────────────────────────────────────────────────────────────────────────────────────────────────┬──────────────────────────────────┬─────────────────────────────────────────────────┬─────────────────────────────────────────────────┬────────┬───────────────┬─────────────────────────────────────────────────┬─────────────────────┬──────────────────────────────────────────────┬────────────┬──────────────────────┐
│ (index) │                                             execution                                             │               type               │                    startTime                    │                    closeTime                    │ status │ historyLength │                  executionTime                  │        memo         │               searchAttributes               │ taskQueue  │ stateTransitionCount │
├─────────┼───────────────────────────────────────────────────────────────────────────────────────────────────┼──────────────────────────────────┼─────────────────────────────────────────────────┼─────────────────────────────────────────────────┼────────┼───────────────┼─────────────────────────────────────────────────┼─────────────────────┼──────────────────────────────────────────────┼────────────┼──────────────────────┤
│    0    │ WorkflowExecution { workflowId: 'my-business-id', runId: '2798482a-46d8-4f1e-ab87-1ba3f7ddda1c' } │ WorkflowType { name: 'example' } │ Timestamp { seconds: [Long], nanos: 125158275 } │ Timestamp { seconds: [Long], nanos: 263021256 } │   2    │    [Long]     │ Timestamp { seconds: [Long], nanos: 125158275 } │ Memo { fields: {} } │ SearchAttributes { indexedFields: [Object] } │ 'tutorial' │        [Long]        │
│    1    │ WorkflowExecution { workflowId: 'my-business-id', runId: '76f1171b-7a73-46a3-ba66-b77bab0b73f8' } │ WorkflowType { name: 'example' } │ Timestamp { seconds: [Long], nanos: 841243925 } │ Timestamp { seconds: [Long], nanos: 935283589 } │   2    │    [Long]     │ Timestamp { seconds: [Long], nanos: 841243925 } │ Memo { fields: {} } │ SearchAttributes { indexedFields: [Object] } │ 'tutorial' │        [Long]        │
│    2    │ WorkflowExecution { workflowId: 'my-business-id', runId: '6d1197b7-41b8-47be-89b4-f1ef3446de1a' } │ WorkflowType { name: 'example' } │ Timestamp { seconds: [Long], nanos: 425778697 } │ Timestamp { seconds: [Long], nanos: 523022091 } │   2    │    [Long]     │ Timestamp { seconds: [Long], nanos: 425778697 } │ Memo { fields: {} } │ SearchAttributes { indexedFields: [Object] } │ 'tutorial' │        [Long]        │
└─────────┴───────────────────────────────────────────────────────────────────────────────────────────────────┴──────────────────────────────────┴─────────────────────────────────────────────────┴─────────────────────────────────────────────────┴────────┴───────────────┴─────────────────────────────────────────────────┴─────────────────────┴──────────────────────────────────────────────┴────────────┴──────────────────────┘
```

```

File: docs/typescript/clients.md [hunk 3]
```diff
@@ -154,6 +155,18 @@ const result = await handle.result(); // block until the workflow completes, if
...
+  ```ts
+   const handle = client.getHandle(workflowId);
+   try {
+    	const result = await handle.result(); // block until the workflow completes, if you wish
+  	} catch (err) {
+  	  if (err instanceOf WorkflowFailedError) {
+  	     throw new Error('Temporal workflow failed: ' + workflowId), err)
+      } else {
+         throw new Error('error from Temporal workflow ' + workflowId), err)
+    }
```
Comment (sw-yx) [inline docs/typescript/clients.md:167]:
```suggestion
   const handle = client.getHandle(workflowId);
   try {
    	const result = await handle.result(); // block until the workflow completes, if you wish
  	} catch (err) {
  	  if (err instanceOf WorkflowFailedError) {
  	     throw new Error('Temporal workflow failed: ' + workflowId, { cause: err})
      } else {
         throw new Error('error from Temporal workflow ' + workflowId, {cause: err})
    }
```

File: docs/typescript/clients.md [hunk 4]
```diff
@@ -246,3 +259,97 @@ export async function CancelExternalWorkflow(wfId: string): void {
 ```
 
 Again, see the [Workflow APIs documentation](/docs/typescript/workflows#external-workflows) for full details.
+
+## Advanced: Making raw gRPC calls
+
+Under the hood of a `WorkflowClient`, the `Connection` is actually powered by a `WorkflowService` driver that makes the raw gRPC calls to Temporal Server.
+This Service is capable of making a wider range of introspection calls (as per [the API reference](https://typescript.temporal.io/api/classes/proto.temporal.api.workflowservice.v1.WorkflowService-1#methods)):
+
+```ts
```
Comment (bergundy) [inline docs/typescript/clients.md:268]: I'd really rather stop putting snippets directly into the docs, we have snipsync, let's use it so we know the snippets work.

File: docs/typescript/clients.md [hunk 5]
```diff
@@ -246,3 +259,97 @@ export async function CancelExternalWorkflow(wfId: string): void {
...
+      workflowTaskStartedEventAttributes: [WorkflowTaskStartedEventAttributes]
+    }
+  ]
+}
+```
+
+</details>
+
+<details>
+<summary><a class="font-mono" href="https://typescript.temporal.io/api/classes/proto.temporal.api.workflowservice.v1.WorkflowService-1#listworkflowexecutions) (requires ElasticSearch">listWorkflowExecutions</a>
```
Comment (bergundy) [inline docs/typescript/clients.md:332]: I don't think this requires elasticsearch if you use basic queries.

**Review by bergundy — APPROVED** [review]
Thanks

File: docs/typescript/clients.md (hunk 5 above)
Comment (sw-yx) [reply docs/typescript/clients.md:332]: i tried it, it does

Comment (sw-yx) [reply docs/typescript/clients.md:332]:
```suggestion
<summary><a class="font-mono" href="https://typescript.temporal.io/api/classes/proto.temporal.api.workflowservice.v1.WorkflowService-1#listworkflowexecutions) (requires ElasticSearch)">listWorkflowExecutions</a>
```

File: docs/typescript/clients.md (hunk 4 above)
Comment (sw-yx) [reply docs/typescript/clients.md:268]: yeah ok ok on it

File: docs/typescript/clients.md [hunk 6]
```diff
@@ -18,6 +18,7 @@ They are the only way to schedule new Workflow Executions with Temporal Server.
   Given a `workflowId`, Workflow Clients may also get a Handle to a running Workflow Execution or retrieve/wait for its result.
 - **Workflow Handles** are bindings to specific Workflow Executions that expose more APIs for control.
   **We strongly recommend familiarising yourself with Workflow Handle APIs** as they are the main way you will signal, query, describe, cancel, terminate and await the result of running Workflow Executions.
+- Advanced users can also use the `WorkflowService` exposed by a Workflow Client to make **raw gRPC calls** (usually for introspection)
```
Comment (aarohib) [inline docs/typescript/clients.md:21]: Add a period at the end of this sentence.

**Comment by flossypurse** [comment]
OK, so we are now at a place where we would like to start enforcing standards around archetypes.
This PR seems like an ideal candidate as it seems to map exactly to a "how to" - i.e. "How to make raw gRPC calls in Typescript".

File: docs/typescript/clients.md [hunk 7]
```diff
@@ -246,3 +263,75 @@ export async function CancelExternalWorkflow(wfId: string): void {
 ```
 
 Again, see the [Workflow APIs documentation](/docs/typescript/workflows#external-workflows) for full details.
+
+## Advanced: Making raw gRPC calls
+
+Under the hood of a `WorkflowClient`, the `Connection` is actually powered by a `WorkflowService` driver that makes the raw gRPC calls to Temporal Server.
+This Service is capable of making a wider range of introspection calls (as per [the API reference](https://typescript.temporal.io/api/classes/proto.temporal.api.workflowservice.v1.WorkflowService-1#methods)):
```
Comment (djmagee) [inline docs/typescript/clients.md:270]:
```suggestion
This Service is capable of making a wider range of introspection calls. (For more information, see the Methods section of the [WorkflowService](https://typescript.temporal.io/api/classes/proto.temporal.api.workflowservice.v1.WorkflowService-1#methods)) API reference.
```

File: docs/typescript/clients.md [hunk 8]
```diff
@@ -245,4 +263,76 @@ export async function CancelExternalWorkflow(wfId: string): void {
 }
 ```
 
-Again, see the [Workflow APIs documentation](/docs/typescript/workflows#external-workflows) for full details.
+Again, see [Workflows in TypeScript](/docs/typescript/workflows#external-workflows) for full details.
+
+## Advanced: Making raw gRPC calls
```
Comment (flossypurse) [inline docs/typescript/clients.md:268]: Any chance you would be willing to convert this to a "how to" page?

**Review by flossypurse — APPROVED** [review]

File: docs/typescript/clients.md (hunk 8 above)
Comment (flossypurse) [reply docs/typescript/clients.md:268]: Go ahead and get this in as-is.

**Pull request merged**